src/main/java/bio/terra/service/dataset/flight/datadelete/DatasetDataDeleteFlight.java

```diff
@@ -35,17 +35,17 @@ public DatasetDataDeleteFlight(FlightMap inputParameters, Object applicationCont
     ApplicationConfiguration appConfig = appContext.getBean(ApplicationConfiguration.class);
 
     // get data from inputs that steps need
-    String datasetId = inputParameters.get(JobMapKeys.DATASET_ID.getKeyName(), String.class);
+    UUID datasetId = JobMapKeys.DATASET_ID.get(inputParameters);
 
     RetryRule lockDatasetRetry =
         getDefaultRandomBackoffRetryRule(appConfig.getMaxStairwayThreads());
 
     addStep(
         new VerifyAuthorizationStep(
-            iamClient, IamResourceType.DATASET, datasetId, IamAction.SOFT_DELETE));
+            iamClient, IamResourceType.DATASET, datasetId.toString(), IamAction.SOFT_DELETE));
```

Comment:
I wonder if there's any way that we could rewrite the `VerifyAuthorizationStep` so that it can also type check the `resourceId` (in this case, the `datasetId`).